File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.go
func timeRequiredToBuy(tickets []int, k int) int {
    continuous := true
    i := 0
    time := 0

    for continuous {
        if i == len(tickets) {
            i = 0
            continue
        }
        if tickets[i] == 0 {
            i++
            continue
        } else {
            tickets[i]--
            time++
        }
        if i == k && tickets[i] == 0 {
            break
        }
        i++
    }
    return time
}

File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.cpp
class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        queue<int> q = {};
        for (int ticket : tickets) {
            q.push(ticket);
        }
        uint16_t timeEllapsed = 0;
        uint8_t kPos = k;
        while (!q.empty()) {
            uint8_t customer = q.front();
            q.pop();
            customer--;
            timeEllapsed++;
            if (customer > 0) {
                q.push(customer);
                if (kPos == 0) {
                    kPos = q.size();
                }
            } else {
                if (kPos == 0) {
                    return timeEllapsed;
                }
            }
            kPos--;
        }
        return -1;
    }
};